pom.xml

```diff
@@ -59,12 +59,16 @@
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 
-        <!-- 16-Oct-2019, tatu: No point trying to target Java 7, Java 8 needed for build anyway -->
         <javac.src.version>1.8</javac.src.version>
         <javac.target.version>1.8</javac.target.version>
 
-        <version.kotlin>1.5.32</version.kotlin>
-
+	<!-- 11-Mar-2023, tatu: [kotlin#632] drops Kotlin 1.4 so target 1.5, but
+	       build with 1.6.x for now
+	  -->
+        <version.kotlin>1.6.21</version.kotlin>
+        <kotlin.compiler.languageVersion>1.5</kotlin.compiler.languageVersion>
+        <kotlin.compiler.apiVersion>1.5</kotlin.compiler.apiVersion>
```

Comment:
I tried a little, but it seems that we need to specify `Kotlin 1.8` as the compiler version in order to generate a build error in a project that uses `Kotlin 1.4`.
I confirmed this by installing `kotlin-stdlib` in a project that specifies `Kotlin 1.4` with different versions.

I also think this approach could cause problems as commented below.
https://github.com/FasterXML/jackson-module-kotlin/pull/623#issuecomment-1435645194

Reply:
Personally, I think it would be better to check the version of `Kotlin` when initializing a `KotlinModule` and throw an error if an unsupported version is used.
This is a little less reliable than a build error, but users of `jackson-module-kotlin` will certainly be aware of it.

Another approach would be to upgrade only the compiler version and continue to use a lower version of `stdlib`, etc.

Reply:
@k163377 +1 for check of actually used Kotlin version when registering module. I am not sure how to do that but I like the idea. I think it'd be orthogonal to question of how to build the module.

If you prefer I remove this part of changes I can do that too -- it is perfectly possible for us to declare 1.4 not supported (and if possible block use with 1.4) without build change beyond 1.5.

What I do want to do is just to get 1.4 deprecation in for Jackson 2.15.0-rc1, which I hope to release within a week.

I think what I'll do is remove all changes except for documentation and C!, and merge those.
Then dependency/build changes can go in different PR?